### PR TITLE
Adding member to a game

### PR DESCRIPTION
**Pull request by andrew-kuzmin**

### CODE REVIEW

File: client/src/app/add-player-dialog/add-player-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    for (const groupMember of this.groupMembers) {
+      if (!(this.currentPlayers.map(value => value.id).indexOf(groupMember.id) >= 0)) {
+        this.playersShowInDialog.push(groupMember);
+      }
+    }
+  }
+
+  addPlayers(list) {
+    this.selectedPlayers = list.selectedOptions.selected.map(item => item.value);
+    console.log(this.selectedPlayers);
```
Comment (JackKarichkovskiy) [inline client/src/app/add-player-dialog/add-player-dialog.component.ts:37]: Please remove

File: client/src/app/game-details/game-details.component.ts [hunk 2]
```diff
@@ -35,4 +39,23 @@ export class GameDetailsComponent implements OnInit {
...
+    const dialogRef = this.dialog.open(AddPlayerDialogViewComponent, {
+      width: '300px',
+      data: {groupId : this.groupId, currentPlayers: this.game.players}
+    });
+
+    dialogRef.afterClosed().subscribe((result: UserProjection[]) => {
+      if (result) {
+        const playersList: GamePlayersList = new GamePlayersList(result.map(value => value.id));
+        this.gameService.addPlayers(this.groupId, this.game.id, playersList).toPromise()
+          .then((newPlayers: UserProjection[]) => this.game.players = newPlayers,
```
Comment (JackKarichkovskiy) [inline client/src/app/game-details/game-details.component.ts:54]: Just clarifying. Should we add new players to the list or replace with new ones?

File: client/src/app/add-player-dialog-view/add-player-dialog-view.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+})
+export class AddPlayerDialogViewComponent implements OnInit {
+
+  $groupMembers: Observable<UserProjection[]>;
+
+  currentPlayers: UserProjection[];
+
+  constructor(
+    public dialogRef: MatDialogRef<AddPlayerDialogViewComponent>,
+    @Optional() @Inject(MAT_DIALOG_DATA) public data: any,
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply client/src/app/add-player-dialog-view/add-player-dialog-view.component.ts:20]: Just got it from a tutorial. No problem, removing it.

File: client/src/app/game-details/game-details.component.ts (hunk 2 above)
Comment (andrew-kuzmin) [reply client/src/app/game-details/game-details.component.ts:54]: I prefer to use the response from the server with an up-to-date game players list. I assume someone else could be also adding players meanwhile, thus a user will have the latest version. What do you think?

Comment (JackKarichkovskiy) [reply client/src/app/game-details/game-details.component.ts:54]: Totally agree, didn't get that gameService.addPlayers returns all merged players

**Pull request merged**